photon-api/src/main/scala/com/linkedin/photon/ml/SparkSessionConfiguration.scala

```diff
@@ -52,6 +52,8 @@ object SparkSessionConfiguration {
     classOf[GeneralizedLinearModel],
     classOf[GeneralizedLinearOptimizationProblem[_]],
     classOf[GLMOptimizationConfiguration],
+    classOf[HessianDiagonalAggregator],
```

Comment:
Why do we need to add this class?

Reply:
I noticed that we had the `ValueAndGradientAggregator` and `HessianVectorAggregator` being registered with Kryo, but not the `HessianDiagonalAggregator` or the `HessianMatrixAggregator`. Since all four function the same way, it stands to reason that they should either all be registered or none of them be registered. Since objects of these four class types will for sure be distributed, therefore they should all be registered.

Apparently, Kryo was able to figure this out on its own, but I think it best to make it explicit.